### ISSUE TITLE
Slow export for scene containing 4000 objects

### ISSUE THREAD

**Issue by rotoglup** [KhronosGroup/glTF-Blender-IO#1687]
When exporting a scene with 4000 objects, with only one of them visible, I noticed that performance is problematic - even when exporting only the visible object.

I came across this problem with a scene containing many variants of kitchen furniture elements, I can't provide this file but I provide a synthetic replacement.

1. Open the attached blend file (containing 4000 cubes)
2. Export as GLB using the following script snippet, to export the single visible cube

```python
import bpy
bpy.ops.export_scene.gltf(filepath=r'4000cubes-b3d_3.2.glb', use_visible=True, use_active_scene=True, export_materials='PLACEHOLDER', export_animations=False, export_morph=False, export_skins=False)
```

3. I obtain the following result (2+ seconds to export the single cube)

```
16:40:43 | INFO: Draco mesh compression is available, use library at C:\nllbin\blender-3.3.0-alpha+master.7f4ee97b9ef9-windows.amd64-release\3.3\python\lib\site-packages\extern_draco.dll
16:40:43 | INFO: Starting glTF 2.0 export
16:40:45 | INFO: Extracting primitive: Cube.1643
16:40:45 | INFO: Primitives created: 1
16:40:45 | INFO: Finished glTF 2.0 export in 2.051008939743042 s
```

**Expected behavior**
The export for this single cube should be faster : I get less that 0.1 second when exporting a blend file with a single cube object.

**.blend file/ .gltf**
[4000cubes-b3d_3.2.blend.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/9151109/4000cubes-b3d_3.2.blend.zip)

**Version**
 - OS: Windows 10 x64
 - Blender Version `3.3-alpha+master.7f4ee97b9ef9`

**Additional context**
I've been profiling the export time using `cProfile` :

```py
import bpy
import cProfile
with cProfile.Profile() as pr:
    bpy.ops.export_scene.gltf(filepath=r'4000cubes-b3d_3.2.glb', use_visible=True, use_active_scene=True, export_materials='PLACEHOLDER', export_animations=False, export_morph=False, export_skins=False)

import pstats
from pstats import SortKey
p = pstats.Stats(pr)
p.sort_stats(SortKey.CUMULATIVE).print_stats(30)
```

and I noticed that :

* a majority of the time is spent in `gltf2_blender_gather_tree.py`
* due to many calls to `blender_object.children`
* `children` is a slow method, the [doc](https://docs.blender.org/api/current/bpy.types.Object.html?highlight=children#bpy.types.Object.children) mentions that it "takes `O(len(bpy.data.objects))` time."

**Comment by scurest** [comment]
ob.children can be precomputed and cached if it's the bottleneck, https://blender.stackexchange.com/questions/229469/how-can-the-linear-time-cost-of-blenders-children-python-property-be-avoided-i

**Comment by rotoglup** [comment]
@julienduroure Thanks ! Can you please tell me in which version of Blender this will be included ? Is there any doc that describes the release process for Blender inclusion ?

**Comment by julienduroure** [comment]
This enhancement is included in Blender 3.3 (currently in beta).
See commit here: https://developer.blender.org/rBA5b2953ad0883223bf8f58cf2bd650891f759bb65

About release process: Every commit here is replicated in Blender repository few minutes after.
The choice of branch it will be replicated depends on 2 different things:
- Blender release process step (see https://wiki.blender.org/wiki/Process/Release_Cycle)
- Severity/risk on changes that are done
